### PR TITLE
refactor write_output_file to merge two invocation paths into one.

### PR DESCRIPTION
**Pull request by pnkfelix**
this is a trivial refactor I did while I was investigating issue #91671.

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/back/write.rs [hunk 1]
```diff
@@ -56,28 +56,27 @@ pub fn write_output_file(
...
-            let dwo_output_c = path_to_c_string(dwo_output);
-            llvm::LLVMRustWriteOutputFile(
-                target,
-                pm,
-                m,
-                output_c.as_ptr(),
-                dwo_output_c.as_ptr(),
-                file_type,
-            )
+        let dwo_output_c;
```
Comment (oli-obk) [inline compiler/rustc_codegen_llvm/src/back/write.rs:65]: 🤣 Wasn't there a twitter thread just yesterday about this pattern?

**Comment by oli-obk** [comment]
r=me with rustfmt run

**Comment by pnkfelix** [comment]
> @bors r- rustfmt is unhappy.

Argh I cannot believe I messed that up. I thought I had my local builds set up to catch this...

**Pull request merged**